cloudbot/util/decorators.py

```diff
@@ -0,0 +1,25 @@
+from time import time
+
+SPAM_LIST = {}
+
+
+def spam_guard(func):
+    def wrapper(*args, **kwargs):
+        nick = kwargs.get('nick', None)  # pull the nick from the args
+        if nick:  # if there is no nick specified for whatever reason, carry on
+            global SPAM_LIST
+            if SPAM_LIST.get(nick, None):
+                SPAM_LIST[nick][0] = time()
+                if time() - SPAM_LIST[nick][0] < SPAM_LIST[nick][1]:
```

Comment:
yeah i accidentally rearranged this. The more correct way to do it will be updated on monday